musicplatform/user/views.py

```diff
@@ -0,0 +1,36 @@
+from django.shortcuts import get_object_or_404
+from knox.auth import AuthToken , TokenAuthentication
+from rest_framework.views import APIView
+from django.http import HttpResponse , JsonResponse
+from .models import User
+from .serializers import UserSerializer
+from rest_framework.parsers import JSONParser
+from rest_framework.response import Response
+from rest_framework.permissions import AllowAny , IsAuthenticated
+
+
+# Create your views here.
+
+
+class api_detail( APIView ):
```

Comment:
and here you are not checking if the request user is the same user you want to update you can just create `permissions.py` file and the following:
```python
class UserPermission(permissions.BasePermission):
    def has_object_permission(self, request, view, obj):
        # Read permissions are allowed to any request,
        # so we'll always allow GET, HEAD or OPTIONS requests.
        if request.method in permissions.SAFE_METHODS:
            return True

        # Instance must have an attribute named `owner`.
        return obj == request.user
```

then in view add this permission to `permission_classes`